src/providers/deno.ts

```diff
@@ -148,6 +149,41 @@ export async function getPackageConfig(
       },
     };
   }
+
+  // If there's a package.json, return that:
+  const pkgJsonUrl = new URL("package.json", pkgUrl);
+  const pkgRes = await fetch(pkgJsonUrl.href, this.fetchOpts);
+  switch (pkgRes.status) {
+    case 200:
+    case 304:
+      return (await pkgRes.json()) as PackageConfig;
+  }
+
+  // If there's a deno.json, read off the top-level imports as deps:
+  const denoJsonUrl = new URL("deno.json", pkgUrl);
```

Comment:
Unfortunately this is absolute not how to use import maps for package management. That would be like using lock files to resolve dependencies instead of the package.json file.

The problem instead is resolving unknown dependencies against the default provider and default registry. In this case it's already defaulting to the `npm` registry, so the only problem we have to fix here is why it's trying to use the deno provider to resolve the npm registry, when it should never be possible to do that.